simulation/simulation.py

```diff
@@ -90,16 +159,72 @@ def calculate_yaw_rate(Cy_f, Cy_r):
     Args:
     Cy_f (float): Cornering stiffness of the front tires in N/rad
     Cy_r (float): Cornering stiffness of the rear tires in N/rad
+    Vx, (float): Current velocity of car in m/s
+    St_a, (float): Current steering angle in radians
+    Trr, (float): Current power from rear right wheel in Nm
+    Trl, (float): Current power from rear left wheel in Nm
 
     Returns:
     float: Yaw rate of the vehicle in radians per second
     """
     # Calculate the yaw rate using the given formula
-    Ku = (lr * m) / (Cy_f * (lf + lr)) - (lf * m) / (Cy_r * (lf + lr))
+    Ku = ((-lf * Cy_f + lr * Cy_r) / (Izz * Vx)) - ((lf**2 * Cy_f + lr**2 * Cy_r) / (Izz * Vx)) + ((lf * Cy_f) / Izz) * St_a + (1 / 0.05 * Izz) * calculate_delta_torque(Trr, Trl)
 
     # Return the calculated yaw rate
     return Ku
 
+def calculate_yaw_moment(Trr, Trl):
+    """
+    Calculates the momentum of how the power ratio of each wheel
+    influences the yaw rate of the car
+
+    Args:
+    Trr, (float): Current power from rear right wheel in Nm
+    Trl, (float): Current power from rear left wheel in Nm
+
+    Returns:
+    float: Yaw moment ratio
+    """
+    # Calculate the yaw moment using the given formula
+    M_V = (Trr - Trl) * Tr
+
+    # Return the calculated yaw moment
+    return M_V
+
+def calculate_lateral_velocity(Vx, St_a):
+    """
+    Calculates the lateral velocity of the car during turns.
+
+    Args:
+    Vx, (float): Current velocity of the car in m/s
+    St_a (float): Steering angle in radians
+
+    Returns:
+    float: Lateral velocity (also known as lateral force)
+    """
+    # Calculate the lateral velocity using the given formula
+    Vy = ((-(Cy_f + Cy_r)/(m * Vx)) + (((-lf * Cy_f + lr * Cy_r)/(m * Vx)) - Vx)) + (Cy_f / (m * Vx)) * St_a
+
+    # Return the calculated lateral velocity
+    return Vy
+
+def calculate_delta_torque(Trr, Trl):
+    """
+    Calculates the difference in torque between each wheel
+
+    Args:
+    Trr, (float): Current power from rear right wheel in Nm
+    Trl, (float): Current power from rear left wheel in Nm
+
+    Returns:
+    float: Difference in torque in Nm
+    """
+    # Calculate the delta torque using the given formula
+    delt_T = (Rw / ((2 * Tr) * Gr)) * calculate_yaw_moment(Trr, Trl)
+
+    #return delt_T
+    return delt_T
+
 # Function for calculating the throttle adjustment required to reach the desired yaw rate
 def calculate_throttle_adjustment(pid_controller, current_yaw_rate, desired_yaw_rate):
```

Comment:
Is throttle adjustment still being used? Or can this be removed?

Reply:
It can be removed. It's not being used anymore for simulation.